### PR TITLE
py/objgenerator: Implement PEP479, StopIteration convs to RuntimeError.

### PR DESCRIPTION
**Pull request by dpgeorge**
This commit implements PEP479 which disallows raising StopIteration inside a generator to signal that it should be finished.  Instead, the generator should simply return when it is complete.

See https://www.python.org/dev/peps/pep-0479/ for details.

It's a clean change, it simplifies handling of exceptions in generators.  But code size is increased a bit (around 20 bytes) due to the introduction of a new exception message (which IMO is necessary otherwise the RuntimeError that is raised is too cryptic without a message).

The majority of changes in this commit are to the tests, to get full coverage, and keep as many of the existing ones as possible.

### CODE REVIEW

File: tests/basics/gen_yield_from.py.exp [hunk 1]
```diff
@@ -1,14 +0,0 @@
-here1
```
Comment (pfalcon) [inline tests/basics/gen_yield_from.py.exp]: Deleting this and other files doesn't mean that the tests will fail with CPython3.4?

**Comment by pfalcon** [comment]
Looks good, thanks for looking into this.

File: tests/basics/gen_yield_from.py.exp (hunk 1 above)
Comment (dpgeorge) [reply tests/basics/gen_yield_from.py.exp]: No, these tests should now test the subset of generator functionality that is the same in CPython 3.4-3.7.  All tests that differ on 3.4 vs 3.5 are moved to generator_pep479.py test.

Comment (dpgeorge) [reply tests/basics/gen_yield_from.py.exp]: Note that Travis uses CPython 3.4.3 and it succeeds ok here.

**Comment by dpgeorge** [comment]
Merged in 3f6ffe059f64b3ebc44dc0bbc63452cb8850702b

**Pull request closed without merging**